### ISSUE TITLE
Potential targets

### ISSUE THREAD

**Issue by larsbrinkhoff** [larsbrinkhoff/xForth#1]
8-bit devices:

- [x] 6502
- [ ] 68HC05/08/RS08
- [x] 8051
- [x] AVR
- [ ] H8/300L
- [x] PIC (just midrange for now)
- [ ] ST6
- [x] STM8

16-bit devices:

- [x] MSP430
- [ ] PIC24 (dsPIC)
- [ ] MAXQ20
- [ ] R8C
- [ ] RL78
- [ ] H8/300H
- [ ] 68HC11/12
- [ ] 68HC16
- [ ] C166/C167/ST10/XE2000

32-bit devices:

- [x] Cortex-M, in particular M0
- [ ] Xtensa (ESP8266)
- [ ] RISC-V
- [ ] PIC32 (MIPS)
- [ ] ARC
- [ ] ColdFire
- [ ] MicroBlaze

**Comment by larsbrinkhoff** [comment]
Rough plan, subject to changes:

- PDP-8: done.
- Cortex-M: done.
- PDP-1
- Xtensa

**Comment by larsbrinkhoff** [comment]
PowerPC doesn't seem to be a hot candidate. Parts are "not for new designs" and the smallest program size is 448K which can easily accomodate a resident compiler.

Can't find any SPARC microcontrollers on DigiKey.  But I live next doors to the makers of the LEON...

**Comment by larsbrinkhoff** [comment]
Historical survey: Microcontrollers sold by [ELFA](https://www.elfa.se/) in 2005:

- 68HC05/08
- 68HC11
- PIC12/16/18
- Basic Stamp/Atom
- ST6
- ST7
- AVR
- SX
- MSP430
- Z8

**Comment by larsbrinkhoff** [comment]
I have to take it back about the PowerPC.  The e200 series seems to be alive and kicking.  Probably on its way out, but still.

**Comment by TG9541** [comment]
Any appetite for C16x, ST10F276, etc?

**Comment by larsbrinkhoff** [comment]
Someone asked about C16x/ST10?  I'm adding targets if there's a chance someone will use them one day. (Or if they seem interesting to me personally.)   I never saw anyone mention C16x or ST10 before.  STM seems to classify it as a legacy product.

Availability of parts and evaluation boards is also a factor to consider.

**Comment by TG9541** [comment]
Hmm.. that was I - until I realized that those were already listed. I once worked a lot with C16x, ST10F269, ST10F276, and I also did some XE2000 coding. The instruction set should be well suited for  a Forth kernel. But you're right, these µCs are a thing of the past, just like GSM phones ;-)

**Comment by larsbrinkhoff** [comment]
I don't see any tiny MIPS devices.  The PIC32 ones are smallish.

This one isn't small except in physical dimensions.  Still nice:  
http://vocore.io/v2.html

**Comment by TG9541** [comment]
Well, I guess it's sufficient to run 256 interactive Forth consoles, with a generous helping of RAM each, concurrently.

**Comment by larsbrinkhoff** [comment]
640KB ought to be enough for anybody.

EDIT: For *ten* anybodies - if they're forthers!

**Comment by TG9541** [comment]
Yes. [My Life in a Terabyte](http://www.theregister.co.uk/2003/05/14/my_life_in_a_terabyte/) is my witness.

**Comment by larsbrinkhoff** [comment]
TMS320 DSPs?

There's a plethora of parts.  I can't even begin to list them.

- C2000 series - 32 and 16 bit fixed point.
- C5000 series - 16-bit.
- C6000 series - VLIW, fixed and floating point.

At least for the C28x 32-bit family, there are inexpensive Launchpad boards.

I already have a C55x board, because I ordered the wrong thing when I was getting an MSP430 Launchpad.

**Comment by larsbrinkhoff** [comment]
There are also Launchpad boards with Cortex-R parts.

**Comment by larsbrinkhoff** [comment]
Good article:  
https://jaycarlson.net/microcontrollers/

Mentions the LC87.

**Comment by sbridger** [comment]
I think that the PIC16F1XXX *enhanced* midrange would be sufficient. They are cheaper than any older midrange parts, more capable, and pretty much completely overlap the pinouts and capabilities. (perhaps with the exception of some 6 pin parts. But what could you connect the tether to?)

The twin pointers and larger stack should make them significantly better forth targets I would think.

Also, unlike many other current processors, some of the 8 and 14 pin parts have such small flash space tethered forth may be the only choice.

**Comment by larsbrinkhoff** [comment]
I already support the original midrange PIC, so I'm guessing (without having checked) that adding the enhanced parts to the zoo wouldn't bee too hard.
